python/fledge/plugins/north/opcuaclient/opcuaclient.py

```diff
@@ -112,78 +112,58 @@ def plugin_reconfigure():
 class OpcuaClientNorthPlugin(object):
     """ North Opcua Client Plugin """
 
-    def __init__(self):
+    def __init__(self, config):
         self.event_loop = asyncio.get_event_loop()
+        self.config = config
 
     async def send_payloads(self, payloads):
         is_data_sent = False
         last_object_id = 0
         num_sent = 0
 
         try:
-            _map = config['map']['value']
-            _LOGGER.debug('processing payloads: {}'.format(payloads))
+            _LOGGER.debug('payloads size: {}'.format(len(payloads)))
+            _map = self.config['map']['value']
             _LOGGER.debug('map: {}'.format(_map))
-
+            url = self.config["url"]["value"]
             for p in payloads:
+                asset_code = p['asset_code']
                 last_object_id = p["id"]
-                if p['asset_code'] in _map:
-                    for datapoint, item in _map[p['asset_code']].items():
+                if asset_code in _map:
+                    for datapoint, item in _map[asset_code].items():
                         if not (item.get('node') is None) and not (item.get('type') is None):
                             if datapoint in p['reading']:
                                 read = dict()
                                 read["value"] = p['reading'][datapoint]
                                 read["type"] = item.get('type')
                                 read["node"] = item.get('node')
                                 read["timestamp"] = p['user_ts']
-                                _LOGGER.debug("Time: %s", str(p['user_ts']))
-
-                                await self._send_payloads(read)
-                                
-                                num_sent += 1
-            _LOGGER.info('payloads sent: {num_sent}'.format(num_sent=num_sent))
+                                await self._send_payloads(url, read)
+                            else:
+                                _LOGGER.warning("{} datapoint is missing in map configuration.".format(datapoint))
```

Comment:
These log messages could result in a very large log file. Perhaps better to not log the same datapoint more than once